app/controllers/venues_controller.rb

```diff
@@ -1,9 +1,19 @@
 class VenuesController < ApplicationController
+
+  before_action :authorize_user, except: [:index, :show]
+
   def index
     @venues = Venue.all
   end
 
   def show
     @venue = Venue.find(params[:id])
   end
+
+  protected
+  def authorize_user
+    if !user_signed_in? || !current_user.admin?
+      raise ActionController::RoutingError.new("Not Found")
+    end
+  end
```

Comment:
FYI the way you have this set up, only an admin will be able to see the venues index or show page. Is that what you were going for?